### PR TITLE
Add missing header ids - plus utility for chunking of html documents

### PR DESCRIPTION
**Pull request by jobergum**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: en/content/idealstate.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
----
+
```
Comment (kkraune) [inline en/content/idealstate.html:1]: unintended?

File: en/jdisc/pluggable-frameworks.html [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 
 
-<h2>Troubleshooting and Configuring the Application</h2>
+<h2 id="Troubleshooting-and- configuring-the-application">Troubleshooting and Configuring the Application</h2>
```
Comment (kkraune) [inline en/jdisc/pluggable-frameworks.html:15]:
```suggestion
<h2 id="troubleshooting-and-configuring-the-application">Troubleshooting and Configuring the Application</h2>
```

File: en/access-logging.html [hunk 3]
```diff
@@ -218,14 +218,12 @@ <h3>Rotation interval</h3>
 </p>
 
 
-<h3>Log retention</h3>
+<h3 id="log-rotation">Log retention</h3>
```
Comment (kkraune) [inline en/access-logging.html:221]:
```suggestion
<h3 id="log-retention">Log retention</h3>
```

File: en/jdisc/pluggable-frameworks.html [hunk 4]
```diff
@@ -32,7 +32,7 @@ <h2>Troubleshooting and Configuring the Application</h2>
 </p>
 
 
-<h3>Importing the Missing Package</h3>
+<h3 id="importing-the-missing-packages">Importing the Missing Package</h3>
```
Comment (kkraune) [inline en/jdisc/pluggable-frameworks.html:35]:
```suggestion
<h3 id="importing-the-missing-package">Importing the Missing Package</h3>
```

File: en/performance/sizing-examples.html [hunk 5]
```diff
@@ -120,7 +120,7 @@ <h2 id="grouped-distribution">Grouped Distribution</h2>
 </p>
 
 
-<h2>9 nodes, 3 groups with 3 nodes per group</h2>
+<h2 id="9-nodes-3-groups">9 nodes, 3 groups with 3 nodes per group</h2>
```
Comment (kkraune) [inline en/performance/sizing-examples.html:123]:
```suggestion
<h2 id="9-nodes-3-groups-with-3-nodes-per-group">9 nodes, 3 groups with 3 nodes per group</h2>
```

File: en/performance/sizing-examples.html [hunk 6]
```diff
@@ -183,7 +183,7 @@ <h2>9 nodes, 3 groups with 3 nodes per group</h2>
 
 
 
-<h2>9 nodes, 9 groups with 1 node per group</h2>
+<h2 id="9-nodes-3-groups">9 nodes, 9 groups with 1 node per group</h2>
```
Comment (kkraune) [inline en/performance/sizing-examples.html:186]:
```suggestion
<h2 id="9-nodes-3-groups-with-1-node-per-group">9 nodes, 9 groups with 1 node per group</h2>
```

File: en/predicate-fields.html [hunk 7]
```diff
@@ -196,7 +196,7 @@ <h4>Identifying Subqueries in Results</h4>
   required to retrieve the subquery bitmap of each hit.
 </p>
 
-<h4>Example</h4>
+<h4 id="predicate-example">Example</h4>
```
Comment (kkraune) [inline en/predicate-fields.html:199]:
```suggestion
<h4 id="predicate-example">Predicate Example</h4>
```

File: en/text-matching-ranking.html [hunk 8]
```diff
@@ -192,7 +192,7 @@ <h4>Array of string attributes</h4>
       Adding "Goldman" and "Sachs" will match query terms like "Gold" and "Sach".
 
 <h3 id="N-Gram-match">N-Gram Match</h3>
-<h4>N-Gram indexing</h4>
+<h4 id="N-Gram-indexing">N-Gram indexing</h4>
```
Comment (kkraune) [inline en/text-matching-ranking.html:195]:
```suggestion
<h4 id="n-gram-indexing">N-Gram indexing</h4>
```

File: en/text-matching-ranking.html [hunk 9]
```diff
@@ -248,7 +248,7 @@ <h4>N-Gram indexing</h4>
     This will enable matching of any 2-term substring, which makes more sense in traditional Chinese than in English.
 </p>
 <h3 id="regexp-match">Regular expression Match</h3>
-<h4>Regular expressions</h4>
+<h4 id="regular expressions">Regular expressions</h4>
```
Comment (kkraune) [inline en/text-matching-ranking.html:251]:
```suggestion
<h4 id="regular-expressions">Regular expressions</h4>
```

**Review by kkraune — APPROVED** [review]
great work!

File: en/content/idealstate.html (hunk 1 above)
Comment (jobergum) [reply en/content/idealstate.html]: Yes, thanks!

**Pull request merged**